keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1001 @@
+
+###### Will enabling / using this feature result in introducing new API types?
+
+A helper API to create CCI Drivers (CCI Driver Factory) can be integrated as staged repo.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No 
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+This will not be true for Alpha.  For Beta, where there may be multiple drivers, it may be prudent to consider this.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No  
+
+### Troubleshooting
+  
+In alpha phase we propose to handle bootstrapping and failures due unavailable drivers 
+by going back to best-effort QoS. In post-alpha phase we will consider solutions of the bootstrapping and driver failure problem
+though an association mechanism between pods and drivers. If a pod does not require a driver CCI Manager can 
+provide allocation for this pod without and driver invocations, which will allow a smooth bootstrapping procedure of the cluster.
+
+If a CCI Driver fails, Pods associated with a driver which were about to be allocated will fail. Based on 
+requirements a fallback mechanism can be implemented where such Pods fallback to 
+a given std. QoS Model. 
+
+![image](CCITroubleshooting.jpg)<br>
+
+Figure 4: Troubleshooting Flows: Handling of pods requiring CCI driver and standard pods wihtout CCI Driver requirements in the case of unhealthy driver.<br><br>
+
+To illustrate this lets consider some of the flows described on Figure 4. We distriguish 
+between pods handled by CCI drivers(group 1 pods) and standard Pods(group 2 pods). Until the CCI driver is not started the pod admission of group 1 pods will fail with an admission error due to driver unavailabilty. Group 2 pods are not impaceted by and they
+can be deployed on the cluster. After the CCI Driver comes online  the admission of 
+group 1 pods can continue. The admission transactions of group 1 and 2 are mutually exclusive. In later time the driver becomes unhealthy and the admission of second pod 
+from group 1 will fail.
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Feature lives completly in kubelet and will not be impacted directly. 
+
+###### What are other known failure modes?
+
+TBD
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+We plan to provide sufficient logging information inside kubelet log to help determine
+problems with CCI Resource Management Stack. Additionally CCI Driver implementations should 
+consider integrating proper logging mechanism and error messages to help determine any 
+issues during operation.
+
+## Challenges and Considerations
+###### Checkpoint assignments
```

Comment:
This is a very fundamental feature and should be dealt with in alpha phase. Making sure that pods recover after kubelet is restarted is very important IMO.